docs/sphinx/user_guide/feature/resource.rst

```diff
@@ -0,0 +1,268 @@
+.. ##
+.. ## Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+.. ## and other RAJA project contributors. See the RAJA/COPYRIGHT file
+.. ## for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+.. ##
+
+.. _resource-label:
+
+=========
+Resources
+=========
+
+This section describes the basic concepts of Resource types and their functionality in 
+``RAJA::forall``. Resources are used as an interface to various backend constructs and their 
+respective hardware. Currently there exists Resource types for ``Cuda``, ``Hip``, ``Omp`` (target) 
+and ``Host``. Resource objects allow the user to execute ``RAJA::forall`` calls asynchronously on a
+respective thread/stream. The underlying concept of each individual Resource is still under 
+development and it should be considered that functionality / behaviour may change.
+
+.. note:: * Currently feature complete asynchronous behaviour and streamed/threaded support is
+            only available when using ``Cuda`` or ``Hip`` resources. 
+          * The ``RAJA::resources`` namespace aliases the ``camp::resources`` namespace.
+
+Each resource has a set of underlying functionality that is synonymous across all resource types.  
+
+ ===================== ===============================================
+ Methods               Brief description
+ ===================== ===============================================
+ get_platform          Returns the underlying camp platform
+                       the resource is associated with.
+ get_event             Return an Event object for the resource from
+                       the last resource call.
+ allocate              Allocate data per the resource's given
+                       backend.
+ deallocate            Deallocate data per the resource's given
```

Comment:
Do we need a note in here stating the the memory dealloc/memset/memcpy methods only work with pointers that correspond to allocations made by the resource?

Reply:
It might be useful, similar to writing something like native CUDA code it's up to the user to manage how and where memory is allocated and how it is later manipulated.